### ISSUE TITLE
Generated xlsx won't open in Excel

### ISSUE THREAD

**Issue by afrowuk** [mk-j/PHP_XLSXWriter#21]
1. If a written cell value contains non-ASCII characters you cannot open the xlsx in Excel.
2. A written cell value such as +542914888847 is written as a numeric value (it passes the filter_var($value, FILTER_VALIDATE_INT) test on line 235) but results in a corrupt xlsx.

I added preg_replace('/[^(\x20-\x7F)]*/', '', $val) in xmlspecialchars() to fix the first problem but I'm not sure the best way to solve the second problem. For now I will treat all values as strings instead.

**Comment by lismansihotang** [comment]
hi, iam use this too. and the output xls won't open. may you help me??

**Comment by mk-j** [comment]
For 1. Non ASCII values should work as long as they are valid UTF-8 (and not null or other garbage characters).  If it detects invalid UTF-8, it logs to STDERR or your web server's log:
    XLSXWriter::log("Error, invalid UTF8 encoding detected.");

For 2. It looks like this is happening on 64 bit systems. I think I have a fix I'll be posting soon.

**Comment by mk-j** [comment]
@lismansihotang Are you setting the output name to .xls? or .xlsx? That could be the issue (go with xlsx).

**Comment by lismansihotang** [comment]
I try .xlsx but still can't..

**Comment by mk-j** [comment]
```
<?php
include_once("xlsxwriter.class.php");

$header = array(
    '1'=>'string',
    '2'=>'string',
);
$data = array(
    array('a','b'),
    array('c','d'),
);
$writer = new XLSXWriter();
$writer->writeSheet($data,'Sheet1');
$writer->writeToFile('example.xlsx');
```
I tried this example out, it worked fine for me.

**Comment by hyunmui** [comment]
I was facing the same problem.
In my case, it was an error by the encoding of the file. It certainly uses method if the encoding of the page that do a output is UTF-8 with BOM, but the error was occur. 

Check the encoding. (In my case, using the Merge tool, the default encoding of Kdiff3 There was UTF-8 with BOM issues for Windows)

Sorry for Google Translater :-)

**Comment by mk-j** [comment]
If you give php xlsx write invalid UTF-8, it will log an error but will not hard fail.  If you give it valid UTF-8 it should be fine.

**Comment by mk-j** [comment]
This should be resolved in version 0.30
